File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float dollar;
    int i = 0, cents = 0;
    
    do
    {
        // Prompt change
        dollar = get_float("What's the change? ");
    }
    while (dollar < 0);

    // Calculate fewest number of coins
    cents = round(dollar * 100);

    // Quarter
    while (cents >= 25)
    {
        cents = cents - 25;
        i++;
    }

    // Dime
    while (cents < 25 && cents >= 10)
    {
        cents = cents - 10;
        i++;
    }

    // Nickel
    while (cents < 10 && cents >= 5)
    {
        cents = cents - 5;
        i++;
    }

    // Penny
    while (cents < 5 && cents >= 1)
    {
        cents = cents - 1;
        i++;
    }

    // Print fewest number of coins
    printf("Total coins: %i\n", i);
}

File: pset6/cash/cash.py
from cs50 import get_float

def main():
    change()

def change():
    # Non negative float
    dollars = get_float("Change owed: ")
    if dollars < 0:
        main()
    else:
        coins = 0
        cents = int(round(dollars * 100))
        while cents >= 25:
            cents = cents - 25
            coins += 1
        while cents < 25 and cents >= 10:
            cents = cents - 10
            coins += 1
        while cents < 10 and cents >= 5:
            cents = cents - 5
            coins += 1
        while cents < 5 and cents >= 1:
            cents = cents - 1
            coins += 1
        print(coins)

main()
